bin/check-all-gpg-keys-for-expiry.rb

```diff
@@ -0,0 +1,348 @@
+      ', Description: ' + @description % self
+  end
+
+  private
+
+  # Return the meaningful word for a specified level
+  def map_number_to_word_level(level)
+    case level
+    when 0
+      'OK'
+    when 1
+      'WARNING'
+    when 2
+      'CRITICAL'
+    when 3
+      'UNKNOWN'
+    end
+  end
+end
+
+# == Description
+#
+# The CheckAllGpgKeysForExpiry class fetches all GPG keys, check their expiry dates and send
+# appropriate alerts to Sensu
+class CheckAllGpgKeysForExpiry < Sensu::Plugin::Check::CLI
+  option :warning,
+         short: '-w NUMBER_OF_DAYS_LEFT_TO_TRIGGER_ALERT',
+         long: '--warning NUMBER_OF_DAYS_LEFT_TO_TRIGGER_ALERT',
+         proc: proc(&:to_i),
+         default: 14,
+         description: 'The minimum number of days left to trigger a warning alert'
+
+  option :critical,
+         short: '-c NUMBER_OF_DAYS_LEFT_TO_TRIGGER_ALERT',
+         long: '--critical NUMBER_OF_DAYS_LEFT_TO_TRIGGER_ALERT',
+         proc: proc(&:to_i),
+         default: 5,
+         description: 'Minimum number of days left to trigger a critical alert'
+
+  option :home_directory,
+         short: '-h GPG_HOME_DIR',
+         long: '--home-directory GPG_HOME_DIR',
+         default: '/root/.gnupg',
+         description: 'Location of GPG home directory'
+
+  option :gpg,
+         short: '-e GPG_EXECUTABLE_PATH',
+         long: '--execute GPG_EXECUTABLE_PATH',
+         default: '/usr/bin/gpg',
+         description: 'Location of binary executable file (gpg)'
+
+  option :use_secret_key,
+         short: '-s (yes|no)',
+         long: '--use-secret-key (yes|no)',
+         default: 'yes',
+         description: 'Check secret key instead of public key'
+
+  option :do_not_display_json_message,
+         short: '-d (yes|no)',
+         long: '--do-not-display-json-message (yes|no)',
+         default: 'no',
+         description: 'Display JSON message instead of sending it to Sensu'
+
+  def initialize
+    super
+  end
+
+  # Return the number of days since epoch from a given epoch
+  def self.days_since_epoch(e)
+    epoch = Date.new(1970, 1, 1)
+    d = Time.at(e).to_date
+    (d - epoch).to_i
+  end
+
+  # Return --list-key if 'use_secret_key' is set to nil or no otherwise
+  # return --list-secret-key
+  def key_type_option
+    if config[:use_secret_key].nil? || config[:use_secret_key] == 'no'
+      '--list-key'
+    else
+      '--list-secret-key'
+    end
+  end
+
+  # Add an option 'home_directory' as an argument to GPG command line if it is
+  # specified.
+  def home_dir_option
+    if config[:home_directory].nil?
+      [config[:gpg]]
+    else
+      [config[:gpg] + " --homedir #{config[:home_directory]}"]
+    end
+  end
+
+  # Return all GPG keys' details by executing the GPG command
+  def fetch_all_gpg_keys
+    args = ['--with-colons', '--fixed-list-mode', '--lock-never', key_type_option]
+    gpg_cmd = home_dir_option + args
+
+    _cmd_out, _cmd_err, _status = Open3.capture3 gpg_cmd.join ' '
```

Comment:
IMHO `gpg_cmd.join(' ')` or `gpg_cmd.join("\s")` is easier to read.

Reply:
This has been done.